### DISCUSSION TITLE
Eager loading aggregate functions of nested relations

### DISCUSSION THREAD

**Discussion by l-alexandrov** [laravel/framework#38049] [Ideas]
I want to eager load some aggregate function  like `withMin` of a nested relation, but it is not possible.

Let's imagine we have the following models:

- Author - hasMany Posts
- Post - hasMany Comments
- Comment

I want to execute the following query:
`Author::withMin('posts.comments', 'rating')->get()`.

**Comment by sbc640964** [comment]
I also want!

**Comment by erikn69** [comment]
Maybe this
```php
Author::withMin('rating')->with(['posts'=>function($query){
    $query->withMin('comments')->groupBy('author_id');
}])->get()
```

**Reply by erikn69** [reply]
@sbc640964 just fix it, i'am on a cellphone, not on an editor
```php
Author::with(['posts'=>function($query){
    $query->withMin('comments','rating')->groupBy('author_id');
}])->get()
```
Maybe better with [has-many-through](https://laravel.com/docs/8.x/eloquent-relationships#has-many-through) comments, example:
```php
Author::withMin('comments','rating')->get()
```

**Reply by sbc640964** [reply]
It works beautifully
But I get all the posts and ratings of everyone

What about I want to get the calculation of the author's rating by calculating the rating of all the posts (without getting all the posts themselves)

**Reply by erikn69** [reply]
@did you try `Author::withMin('comments','rating')->get()`, seems to be closer to what you are looking for

**Reply by sbc640964** [reply]
The problem is that the relationship of the comments is to the posts and not to the authors

**Reply by erikn69** [reply]
>Maybe better with [has-many-through](https://laravel.com/docs/8.x/eloquent-relationships#has-many-through) comments, example:

did you read the documentation?

**Comment by sbc640964** [comment]
@erikn69
It can not work.
withMin must have two parameters (relationship and column)
